### PR TITLE
CB-17983 , CB-18036 Several fix from the backlog

### PR DESCRIPTION
**Pull request by doktoric**
See detailed description in the commit message.

### CODE REVIEW

File: core/src/test/java/com/sequenceiq/cloudbreak/structuredevent/rest/urlparsers/RestUrlParserTest.java [hunk 1]
```diff
@@ -92,7 +92,7 @@ public class RestUrlParserTest {
             "cluster_templates", "/v4/events", "/v4/diagnostics", "/v4/progress", "/v4/operation",
             "/v4/custom_configurations"};
 
-    private final String[] excludePaths = {"/stacks/internal/crn"};
+    private final String[] excludePaths = {"/stacks/internal/crn", "/govCloud"};
```
Comment (lajosrodek) [inline core/src/test/java/com/sequenceiq/cloudbreak/structuredevent/rest/urlparsers/RestUrlParserTest.java:95]: Nit (and unrelated to the present changes): `excludes` and `excludePaths` should rather by constants, i.e. `private static final String[] EXCLUDES` and `EXCLUDE_PATHS`.

File: environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentModificationService.java [hunk 2]
```diff
@@ -357,6 +372,34 @@ private void editAdminGroupNameIfChanged(Environment environment, EnvironmentEdi
...
+        NetworkDto networkDto = environmentDtoConverter.networkToNetworkDto(environment);
+        AddDnsZoneForSubnetIdsRequest addDnsZoneForSubnetIdsRequest = addDnsZoneForSubnetIdsRequest(editDto, environment, networkDto);
+        if (shouldSendSubnetIdsToFreeIpa(addDnsZoneForSubnetIdsRequest)) {
+            dnsV1Endpoint.addDnsZoneForSubnetIds(addDnsZoneForSubnetIdsRequest);
+        }
+    }
+
+    private AddDnsZoneForSubnetIdsRequest addDnsZoneForSubnetIdsRequest(EnvironmentEditDto environmentEditDto,
+        Environment environment, NetworkDto networkDto) {
+        AddDnsZoneForSubnetIdsRequest addDnsZoneForSubnetIdsRequest = new AddDnsZoneForSubnetIdsRequest();
```
Comment (lajosrodek) [inline environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentModificationService.java:390]: Wrong indentation in the second line of the constructor header.

File: environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentModificationService.java [hunk 3]
```diff
@@ -357,6 +372,34 @@ private void editAdminGroupNameIfChanged(Environment environment, EnvironmentEdi
         }
     }
 
+    private boolean shouldSendSubnetIdsToFreeIpa(AddDnsZoneForSubnetIdsRequest addDnsZoneForSubnetIdsRequest) {
+        AddDnsZoneNetwork addDnsZoneNetwork = addDnsZoneForSubnetIdsRequest.getAddDnsZoneNetwork();
+        return StringUtils.isNotBlank(addDnsZoneNetwork.getNetworkId()) && !addDnsZoneNetwork.getSubnetIds().isEmpty();
```
Comment (lajosrodek) [inline environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentModificationService.java:377]: NPE if `addDnsZoneNetwork.getSubnetIds() == null` (which is possible according to `addDnsZoneForSubnetIdsRequest()`, and the field has no explicit initializer).

File: environment/src/main/java/com/sequenceiq/environment/environment/dto/EnvironmentDtoConverter.java [hunk 4]
```diff
@@ -204,6 +205,15 @@ public Environment creationDtoToEnvironment(EnvironmentCreationDto creationDto)
         return environment;
     }
 
+    public NetworkDto networkToNetworkDto(Environment environment) {
```
Comment (lajosrodek) [inline environment/src/main/java/com/sequenceiq/environment/environment/dto/EnvironmentDtoConverter.java:208]: Could you please cover with UT?

File: environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentModificationService.java [hunk 5]
```diff
@@ -161,6 +175,7 @@ private EnvironmentDto edit(EnvironmentEditDto editDto, Environment env) {
         editCloudStorageValidation(editDto, env);
         editTunnelIfChanged(editDto, env);
         editEnvironmentParameters(editDto, env);
+        editFreeIPA(editDto, env);
```
Comment (lajosrodek) [inline environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentModificationService.java:178]: Could you please cover with UT?

**Review by lajosrodek — CHANGES_REQUESTED** [review]

File: environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentModificationService.java [hunk 6]
```diff
@@ -74,11 +78,20 @@ public class EnvironmentModificationService {
 
     private final AzureParametersRepository azureParametersRepository;
 
-    public EnvironmentModificationService(EnvironmentDtoConverter environmentDtoConverter, EnvironmentService environmentService,
-            CredentialService credentialService, NetworkService networkService, AuthenticationDtoConverter authenticationDtoConverter,
-            ParametersService parametersService, EnvironmentFlowValidatorService environmentFlowValidatorService,
-            EnvironmentResourceService environmentResourceService, EnvironmentEncryptionService environmentEncryptionService,
-            AzureParametersRepository azureParametersRepository) {
+    private final DnsV1Endpoint dnsV1Endpoint;
```
Comment (bergerdenes) [inline environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentModificationService.java:81]: it is not used (yet?)

Comment (lajosrodek) [reply environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentModificationService.java:81]: I do not get it. :smile: From `editFreeIPA()`:
```
            dnsV1Endpoint.addDnsZoneForSubnetIds(addDnsZoneForSubnetIdsRequest);
```

File: core/src/test/java/com/sequenceiq/cloudbreak/structuredevent/rest/urlparsers/RestUrlParserTest.java [hunk 7]
```diff
@@ -88,11 +88,11 @@ public class RestUrlParserTest {
     @Autowired
     private List<LegacyRestUrlParser> restUrlParsers;
 
-    private final String[] excludes = {"/v1/distrox", "/v1/internal/distrox", "/flow-public", "/autoscale",
+    private final String[] EXCLUDES = {"/v1/distrox", "/v1/internal/distrox", "/flow-public", "/autoscale",
             "cluster_templates", "/v4/events", "/v4/diagnostics", "/v4/progress", "/v4/operation",
             "/v4/custom_configurations"};
 
-    private final String[] excludePaths = {"/stacks/internal/crn", "/govCloud"};
+    private final String[] EXCLUDEPATHS = {"/stacks/internal/crn", "/govCloud"};
```
Comment (lajosrodek) [inline core/src/test/java/com/sequenceiq/cloudbreak/structuredevent/rest/urlparsers/RestUrlParserTest.java:95]: Missing `static`. :smile:

File: environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentModificationService.java (hunk 2 above)
Comment (lajosrodek) [reply environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentModificationService.java:392]: Still present (though not an error by itself).

File: environment/src/test/java/com/sequenceiq/environment/environment/dto/EnvironmentDtoConverterTest.java [hunk 8]
```diff
@@ -65,6 +66,8 @@
 import com.sequenceiq.environment.tags.service.DefaultInternalAccountTagService;
 import com.sequenceiq.environment.tags.v1.converter.AccountTagToAccountTagResponsesConverter;
 
+import reactor.core.support.Assert;
```
Comment (lajosrodek) [inline environment/src/test/java/com/sequenceiq/environment/environment/dto/EnvironmentDtoConverterTest.java:69]: Not needed (and better to use JUnit 5 `Assertions` or AssertJ anyway). :smile:

File: environment/src/test/java/com/sequenceiq/environment/environment/service/EnvironmentModificationServiceTest.java [hunk 9]
```diff
@@ -805,6 +813,45 @@ void testUpdateAzureResourceEncryptionParametersNoErrorWhenSameEncryptionKeyAlre
...
+
+        EnvironmentEditDto environmentDto = EnvironmentEditDto.builder()
+                .withAccountId(ACCOUNT_ID)
+                .withNetwork(
+                        NetworkDto.builder()
+                                .withNetworkId("abs-123")
+                                .withSubnetMetas(Map.of("subnet-1", new CloudSubnet()))
+                                .build())
+                .build();
+        Environment value = new Environment();
```
Comment (lajosrodek) [inline environment/src/test/java/com/sequenceiq/environment/environment/service/EnvironmentModificationServiceTest.java:828]: Nit: perhaps rename to `environment`.

File: environment/src/test/java/com/sequenceiq/environment/environment/service/EnvironmentModificationServiceTest.java [hunk 10]
```diff
@@ -805,6 +813,45 @@ void testUpdateAzureResourceEncryptionParametersNoErrorWhenSameEncryptionKeyAlre
...
+                .thenReturn(new EnvironmentDto());
+        when(environmentDtoConverter.networkToNetworkDto(any()))
+                .thenReturn(NetworkDto.builder().withNetworkId("abs-123").build());
+
+        environmentModificationServiceUnderTest.editByName(ENVIRONMENT_NAME, environmentDto);
+
+        verify(environmentService, times(1)).findByNameAndAccountIdAndArchivedIsFalse(any(), anyString());
+        verify(networkService, times(1)).validate(any(), any(), any());
+        verify(networkService, times(1)).refreshMetadataFromCloudProvider(any(), any(), any());
+        verify(dnsV1Endpoint, times(1)).addDnsZoneForSubnetIds(any());
```
Comment (lajosrodek) [inline environment/src/test/java/com/sequenceiq/environment/environment/service/EnvironmentModificationServiceTest.java:848]: Could you please capture the `AddDnsZoneForSubnetIdsRequest` arg and then verify its contents?

**Review by lajosrodek — CHANGES_REQUESTED** [review]

File: environment/src/test/java/com/sequenceiq/environment/environment/service/EnvironmentModificationServiceTest.java (hunk 10 above)
Comment (doktoric) [reply environment/src/test/java/com/sequenceiq/environment/environment/service/EnvironmentModificationServiceTest.java:848]: The chain what matter here I think

Comment (lajosrodek) [reply environment/src/test/java/com/sequenceiq/environment/environment/service/EnvironmentModificationServiceTest.java:848]: I feel it is also important that `DnsV1Endpoint` is called with the right arguments. :smile:

**Review by lajosrodek — APPROVED** [review]
Thanks, LGTM. Only one minor note, but approving nevertheless.

**Pull request merged**